tests/e2e/did/switch-controller-flow.spec.ts

```diff
@@ -0,0 +1,445 @@
+import { ID_TYPE, CONTENT_TYPE } from '../constants';
+import { test, expect, APIRequestContext } from '@playwright/test';
+import { StatusCodes } from 'http-status-codes';
+
+import { CheqdNetwork, DIDDocument, VerificationMethods } from '@cheqd/sdk';
+
+import { CreateDidResponseBody, UpdateDidResponseBody } from '@cheqd/credential-service/src/types/did';
+
+test.use({ storageState: 'playwright/.auth/user.json' });
+
+// The flow is:
+// 1. Create a DID which will be used as an original controller (UUID) . DID-A
+// 2. Create a DID which will be used as a new controller (base58). DID-B
+// 3. Create a DID which will be used as another new controller (base58). DID-C
+// 4. Update the original controller DID with the new controller DID. Adding the new one. Expected amount of controllers: 2. DID-A and DID-B
+//    4.1 We expect here to see the list of keys, included all possible keys which can be used for signing.
+//        Also, each element of keys should include `controller` field which will point to the DID with it associated to.
+//    4.2 We expect here in response to see the list of controller keyRefs, which are required for signing.
+// 5. Create a resource for DID-A.
+// 6. Create a resource for DID-B.
+// 7. Update the original controller DID with the controller DID-B. Removing the original one. Expected amount of controllers: 1. DID-B
+// 8. Create a resource using DID-A.
+// 9. Create a resource using DID-B.
+// 10. Update controller to the original DID-A.
+//    (Here we wanna to check that DID with replaced controller could be used as a controller for completly another DID). Expected amount of controllers: 1. DID-A
+// 11. Create a resource using DID-C.
+// 12. Deactivate DID-C.
+// 13. Try to set DID-C as a controller for DID-A. Expected error.
+// 14. Try to send updated DID Document with empty verificationMEthod list. Expected error.
+// 15. Return back te original controller for DID_A Expected amount of controllers: 1
+// 16. Create a resource using DID-A, DID-B.
+
+// Structures
+type ResponseBody = CreateDidResponseBody | UpdateDidResponseBody;
+let DID_A: ResponseBody;
+let DID_B: ResponseBody;
+let DID_C: ResponseBody;
+
+let DIDDocument_A: DIDDocument;
+let DIDDocument_B: DIDDocument;
+let DIDDocument_C: DIDDocument;
+
+// Helpers
+async function searchDID(did: string, request: APIRequestContext): Promise<DIDDocument> {
+	const response = await request.get(`/did/search/${did}`);
+	expect(response).toBeOK();
+
+	return await response.json().then((result) => result.didDocument);
+}
+
+async function isDeactivated(did: string, request: APIRequestContext): Promise<boolean> {
+	const response = await request.get(`/did/search/${did}`);
+	expect(response).toBeOK();
+
+	return await response.json().then((result) => JSON.parse(result.didDocumentMetadata.deactivated));
+}
+
+function mustIncludeKey(didResponse: ResponseBody, keyRef: string): boolean {
+	return didResponse.controllerKeys.some((key) => key.kid === keyRef);
+}
```

Comment:
The naming convention is ambiguous here.

The expected `must<perform-action>` literal indicates the closure throws if the operation fails, which is not the case here.

I appreciate this is a test helper, therefore won't probably confuse end users, hence why marking as non-blocking to promote.